lib/src/components/list_item/list_item.dart

```diff
@@ -2,205 +2,224 @@ import 'package:flutter/foundation.dart';
-    required this.title,
-    this.dense = false,
-    this.enabled = true,
-    this.enabledDivider = true,
+    required this.primaryText,
+    this.secondaryText,
     this.leading,
     this.onTap,
-    this.selected = false,
-    this.subtitle,
+    this.showDivider,
     this.trailing,
     super.key,
   });
 
-  /// Dense list items have less space between widgets and use smaller [TextStyle]
-  final bool dense;
+  /// Creates a [ZetaListItem] with a [ZetaSwitch] in the trailing widget space.
+  ZetaListItem.toggle({
+    super.key,
+    required this.primaryText,
+    this.secondaryText,
+    this.showDivider,
+    this.leading,
+    bool value = false,
+    ValueChanged<bool?>? onChanged,
+  })  : trailing = ZetaSwitch(
+          value: value,
+          onChanged: onChanged,
+          variant: ZetaSwitchType.android,
+        ),
+        onTap = (() => onChanged?.call(!value));
 
-  /// Whether this [ZetaListItem] is interactive.
-  /// If false the [onTap] callback is inoperative.
-  final bool enabled;
+  /// Creates a [ZetaListItem] with a [ZetaCheckbox] in the trailing widget space.
+  ZetaListItem.checkbox({
+    super.key,
+    required this.primaryText,
+    this.secondaryText,
+    this.leading,
+    this.showDivider,
+    bool value = false,
+    bool rounded = true,
+    ValueChanged<bool>? onChanged,
+    bool useIndeterminate = false,
+  })  : trailing = ZetaCheckbox(
+          value: value,
+          onChanged: onChanged,
+          useIndeterminate: useIndeterminate,
+          rounded: rounded,
+        ),
+        onTap = (() => onChanged?.call(!value));
 
-  /// Whether to apply divider. Normally at the bottom of the [ZetaListItem].
-  final bool enabledDivider;
+  /// Creates a [ZetaListItem] with a [ZetaRadio] in the trailing widget space.
+  ZetaListItem.radio({
+    required this.primaryText,
+    required dynamic value,
+    this.secondaryText,
+    this.leading,
+    this.showDivider,
+    dynamic groupValue,
+    super.key,
+    ValueChanged<dynamic>? onChanged,
+  })  : trailing = ZetaRadio<dynamic>(
+          value: value,
+          groupValue: groupValue,
+          onChanged: onChanged,
+        ),
+        onTap = (() => onChanged?.call(value));
 
   /// A Widget to display before the title;
   final Widget? leading;
 
   /// Called when user taps on the [ZetaListItem].
   final VoidCallback? onTap;
 
-  /// Applies selected styles. If selected is true trailing mu
-  final bool selected;
-
-  /// Additional content displayed over the title.
-  /// Typically a [Text] widget.
-  final Widget? subtitle;
+  /// The primary text of the [ZetaListItem].
+  final String primaryText;
 
-  /// The primary content of the [ZetaListItem].
-  final Widget title;
+  /// The primary text of the [ZetaListItem].
+  final String? secondaryText;
 
   /// A widget to display after the title.
   final Widget? trailing;
 
+  /// Adds a border to the bottom of the list item.
+  final bool? showDivider;
```

Comment:
should we add a note here about this being overridden by the `ZetaList` parent?

Reply:
It doesn't get overridden, the showDivider prop takes precedent, but I'll add a note about that